pygen_src/riscv_instr_pkg.py

```diff
@@ -0,0 +1,1595 @@
+    R_W_EXECUTE_PAGE = 0b111   #Read-write-execute page
+
+
+class interrupt_cause_t(Enum):
+    U_SOFTWARE_INTR = 0x0
+    S_SOFTWARE_INTR = 0x1
+    M_SOFTWARE_INTR = 0x3
+    U_TIMER_INTR = 0x4
+    S_TIMER_INTR = 0x5
+    M_TIMER_INTR = 0x7
+    U_EXTERNAL_INTR = 0x8
+    S_EXTERNAL_INTR = 0x9
+    M_EXTERNAL_INTR = 0xB
+
+
+
+class exception_cause_t(Enum):
+    INSTRUCTION_ADDRESS_MISALIGNED = 0x0
+    INSTRUCTION_ACCESS_FAULT = 0x1
+    ILLEGAL_INSTRUCTION = 0x2
+    BREAKPOINT = 0x3
+    LOAD_ADDRESS_MISALIGNED = 0x4
+    LOAD_ACCESS_FAULT = 0x5
+    STORE_AMO_ADDRESS_MISALIGNED = 0x6
+    STORE_AMO_ACCESS_FAULT = 0x7
+    ECALL_UMODE = 0x8
+    ECALL_SMODE = 0x9
+    ECALL_MMODE = 0xB
+    INSTRUCTION_PAGE_FAULT = 0xC
+    LOAD_PAGE_FAULT = 0xD
+    STORE_AMO_PAGE_FAULT = 0xF
+
+
+
+class misa_ext_t(Enum):
+    MISA_EXT_A = 0
+    MISA_EXT_B = auto()
+    MISA_EXT_C = auto()
+    MISA_EXT_D = auto()
+    MISA_EXT_E = auto()
+    MISA_EXT_F = auto()
+    MISA_EXT_G = auto()
+    MISA_EXT_H = auto()
+    MISA_EXT_I = auto()
+    MISA_EXT_J = auto()
+    MISA_EXT_K = auto()
+    MISA_EXT_L = auto()
+    MISA_EXT_M = auto()
+    MISA_EXT_N = auto()
+    MISA_EXT_O = auto()
+    MISA_EXT_P = auto()
+    MISA_EXT_Q = auto()
+    MISA_EXT_R = auto()
+    MISA_EXT_S = auto()
+    MISA_EXT_T = auto()
+    MISA_EXT_U = auto()
+    MISA_EXT_V = auto()
+    MISA_EXT_W = auto()
+    MISA_EXT_X = auto()
+    MISA_EXT_Y = auto()
+    MISA_EXT_Z = auto()
+
+
+class hazard_e(Enum):
+    NO_HAZARD = auto()
+    RAW_HAZARD = auto()
+    WAR_HAZARD = auto()
+    WAW_HAZARD = auto()
+
+import riscv_core_setting
+#import riscv_core_setting
+
+# PMP address matching mode
+class pmp_addr_mode_t(Enum):
+    OFF = 0b00
+    TOR = 0b01
+    NA4 = 0b10
+    NAPOT = 0b11
+
+#PMP configuration register layout
+#This configuration struct includes the pmp address for simplicity
+#TODO (udinator) allow a full 34 bit address for rv32?
+
+
+ 
+#`ifdef _VCP // GRK958
+
+if _VCP:
+    class pmp_cfg_reg_t(Enum):   # struct packed
+        l = auto()
+	zero = auto()
+	a = auto()          #  pmp_addr_mode_t  a;
+	x = auto()
+	w = auto()
+	r = auto()
+	addr = auto()      #bit [XLEN - 1 : 0]    addr;    
+	offset = auto()    #bit [XLEN - 1 : 0]    addr;
+
+#RV32: the pmpaddr is the top 32 bits of a 34 bit PMP address
+#RV64: the pmpaddr is the top 54 bits of a 56 bit PMP address
+#The offset from the address of <main> - automatically populated by the PMP generation routine.
+
+else:
+    class pmp_cfg_reg_t(Enum):   # struct
+        l = randint(0,)
+	zero = auto()
+	a = randint(0,)     # pmp_addr_mode_t a;
+	x = randint(0,)
+	w = randint(0,)
+	r = randint(0,)
+	addr = auto()
+	offset = auto()
+
+
+
+def hart_prefix(hart = 0):
+    if NUM_HARTS <= 1:
+        return ""
+    else:
+        return print("hart")
+
+
+
+
+
+def get_label(label, hart = 0):
+    return {hart_prefix(hart), label}   
+
+
+
+
+class vtype_t(Enum):
+    ill = auto() 

```

Comment:
In SV this vtype_t is taken as a struct . So it's wrong conversion.
FYI
  typedef struct packed {
    bit ill;
    bit [XLEN-2:7] reserved;
    bit [1:0] vediv;
    bit [2:0] vsew;
    bit [1:0] vlmul;
  } vtype_t;

Reply:
Please provide me the python logic for above code...!